.gitattributes

```diff
@@ -0,0 +1 @@
+package-lock.json binary
```

Comment:
What is this line for?  I've not come across this before, curious to learn!

Reply:
> `package-lock.json` is **automatically generated** for any operations where npm modifies either the `node_modules` tree, or `package.json.` It describes the exact tree that was generated, such that subsequent installs are able to generate identical trees, regardless of intermediate dependency updates.

The __emphasis is mine__ above. Because it's automatically generated, it shouldn't show up in `git add --patch` or `git diff` commands since it's essentially not to be edited by humans. [This `.gitattribute` file adds a flag of `binary` to any pathname matching the first part](https://www.git-scm.com/docs/gitattributes). It's useful for other things like saying that a `.doc[x]` file is a text file or a particular path contains different line endings. _ 🙄 at you AppleScript_